pkg/repositories/transformers/task_execution.go

```diff
@@ -152,6 +157,47 @@ func mergeLogs(existing, latest []*core.TaskLog) []*core.TaskLog {
 	return logs
 }
 
+func mergeCustom(existing, latest *_struct.Struct) (*_struct.Struct, error) {
+	if existing == nil {
+		return latest, nil
+	}
+	if latest == nil {
+		return existing, nil
+	}
+
+	// To merge latest into existing we first create a patch object that consists of applying changes from latest to
+	// an empty struct. Then we apply this patch to existing so that the values changed in latest take precedence but
+	// barring conflicts/overwrites the values in existing stay the same.
+	var empty _struct.Struct
+	jsonEmpty, err := json.Marshal(&empty)
```

Comment:
Should we use jsonpb to marshal proto messages?

Reply:
@EngHabu done

Reply:
@EngHabu done